eng/common/docgeneration/Generate-DocIndex.ps1

```diff
@@ -114,10 +117,20 @@ function GenerateDocfxTocContent([Hashtable]$tocContent, [String]$lang, [String]
     New-Item -Path $YmlPath -Name "toc.yml" -Force
     $visitedService = @{}
     # Sort and display toc service name by alphabetical order, and then sort artifact by order.
-    foreach ($serviceMapping in ($tocContent.GetEnumerator() | Sort-Object Value, Key)) {
+    # Track 2 packages go first, then others.
+    # Define the order of "New", "Type"
+    $CustomNewOrder = "true", "false", ""
+    $CustomTypeOrder = "client", "mgmt", "compat", "spring", ""
+    $sortedToc = $tocContent.GetEnumerator() | Sort-Object {
+        $CustomNewOrder.IndexOf($_.Value.New), 
+        $CustomTypeOrder.IndexOf($_.Value.Type), 
+        $_.Value.ServiceName, 
+        $_.Value.DisplayName,
+        $_.Key}
```

Comment:
This sorting looks a little more complex then it should be. I suggest something like:
```suggestion
    $sortedToc = $tocContent.Values | Sort-Object New, Type, ServiceName, DisplayName, Artifact
```

Of course this means you need to add Artifact to the property set of the custom object. Then you can change the below code to this as well. 

```
        $artifact = $serviceMapping.Artifact
        $serviceName = $serviceMapping.ServiceName
        $displayName = $serviceMapping.DisplayName
```

Reply:
It is not there. We do not loop into the mapping yet when sort this.

Reply:
I'm not sure I understand your comment here? If you do $tocContents.Values.

Reply:
I thought we asked me define artifact right here. Sort by key means sort by artifact.